app/src/main/java/doit/study/droid/splash/SplashViewModel.kt

```diff
@@ -0,0 +1,97 @@
+package doit.study.droid.splash
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import doit.study.droid.data.local.QuizContentVersion
+import doit.study.droid.data.local.QuizDatabase
+import doit.study.droid.data.local.entity.Question
+import doit.study.droid.data.local.entity.QuestionTagJoin
+import doit.study.droid.data.local.entity.Tag
+import doit.study.droid.data.remote.Configuration
+import doit.study.droid.data.remote.QuizData
+import doit.study.droid.data.remote.QuizDataClient
+import doit.study.droid.utils.Event
+import kotlinx.coroutines.*
+import timber.log.Timber
+import javax.inject.Inject
+
+class SplashViewModel @Inject constructor(
+        private val quizContentVersion: QuizContentVersion,
+        private val quizDataClient: QuizDataClient,
+        private val quizDatabase: QuizDatabase
+): ViewModel() {
+
+    private val cachedTags = mutableMapOf<String, Int>()
+    private lateinit var config: Configuration
+
+    private val _navigateToTopics = MutableLiveData<Event<Unit>>()
+    val navigateToTopics: LiveData<Event<Unit>> = _navigateToTopics
+
+    private val _showErrorAndExit = MutableLiveData<Event<String>>()
+    val showErrorAndExit: LiveData<Event<String>> = _showErrorAndExit
+
+
+    fun syncWithServer() {
+        GlobalScope.launch(Dispatchers.Main) {
+            try {
+                withContext(Dispatchers.IO) {
+                    if (isThereNewContent()) {
+                        updateData(quizDataClient.quizData())
+                    }
+                }
+            } catch (e: Exception) {
+                _showErrorAndExit.value = Event("Error to sync, try again later")
+                Timber.e(e)
+            } finally {
+                _navigateToTopics.value = Event(Unit)
+            }
+        }
```

Comment:
Can be simpler:
```
viewModelScope.launch {
    try {
        if (isThereNewContent()) {
            updateData(quizDataClient.quizData())
        }
    } catch (e: Exception) {
      … 
    } finally {
       …
    }
}
```

Reply:
BTW, GlobalScope was used to force potential leaking a bit because synchronization here is mandatory and it's better to finish before potential exit.
Can you suggest another strategy in that case?